code/datums/components/construction.dm

```diff
@@ -33,8 +33,9 @@
 
 /datum/component/construction/proc/action(datum/source, obj/item/I, mob/living/user)
 	SIGNAL_HANDLER
-
-	return INVOKE_ASYNC(src, PROC_REF(check_step), I, user)
+	spawn(-1) //Please stop trying to get the return value from procs that can sleep
+		. = check_step(I, user)
```

Comment:
`check_step` should probably `set waitfor = FALSE` instead, then we can `return check_step(I, user)` directly

Reply:
SHOULD_NOT_SLEEP() is a weird lint. It is EXTREMELY aggressive and ignores `set waitfor = FALSE`, it would require an UNLINT() wrap which is uglier imo.

Reply:
> and ignores set waitfor = FALSE

I required they do this because set waitfor = FALSE is often used on should not sleep procs as a backup. should not sleep is a compile time lint of a runtime typed language, there are codepaths that can come up at runtime that do not trigger runtime errors that can cause different procs to run then the compile time code might have been expecting. So any proc called by the mc has to have set waitfor = FALSE, regardless of should not sleep status, just incase.

Reply:
Fair play.